extensions/modules/android-utils/android-utils.lcb

```
@@ -84,9 +84,152 @@ Example:
+-   "android.permission.ANSWER_PHONE_CALLS": permission to allow an application to answer an incoming phone call.
+-   "android.permission.READ_CALL_LOG": permission to allow an application to read the user's call log.
+-   "android.permission.WRITE_CALL_LOG": permission to allow an application to write to the user's call log.
+-   "android.permission.ADD_VOICEMAIL": permission to allow an application to add voicemails into the system.
+-   "android.permission.USE_SIP": permission to allow an application to use SIP service.
+-   "android.permission.PROCESS_OUTGOING_CALLS": permission to allow an application to see the number being dialed during an outgoing call with the option to redirect the call to a different number or abort the call altogether.
+-   "android.permission.SEND_SMS": permission to allow an application to send SMS messages.
+-   "android.permission.RECEIVE_SMS": permission to allow an application to receive SMS messages
+-   "android.permission.READ_SMS": permission to allow an application to read SMS messages.
+-   "android.permission.RECEIVE_WAP_PUSH": permission to allow an application to receive WAP push messages.
+-   "android.permission.RECEIVE_MMS": permission to allow an application to receive MMS messages.
+-   "android.permission.BODY_SENSORS": permission to allow an application to access data from sensors that the user uses to measure what is happening inside his/her body, such as heart rate.
+
+Returns(boolean):
+True if permission has been granted, false otherwise.
+
+Description:
+Use the <AndroidRequestPermission> command to request permission for
+<pPermission> from the user.
+
+>*Note:* Permission names are case sensitive.
+*/
+public handler AndroidRequestPermission(in pPermission as String) returns Boolean
+	variable tHasPermission as Boolean
+	unsafe
+		MCAndroidCheckRuntimePermission(pPermission)
```

Sorry @livecodemichael what I meant was `put MCAndroidCheckRuntimePermission(pPermission) into tHasPermission`

---

Interestingly the LCS variant doesn't have a granted result instead it sets the result to empty even though the granted status is right there.... odd... @livecodepanos do you recall the reasoning for this in LCS? Perhaps the same argument applies here.

---

> Sorry @livecodemichael what I meant was `put MCAndroidCheckRuntimePermission(pPermission) into tHasPermission`

Duh yeah. As you say, the LCS version doesn't return granted result so I assumed the internal function didn't.

---

> Interestingly the LCS variant doesn't have a granted result instead it sets the result to empty even though the granted status is right there.... odd... @livecodepanos do you recall the reasoning for this in LCS? Perhaps the same argument applies here.

It was @runrevmark 's suggestion to return empty. See this comment:
https://github.com/livecode/livecode/pull/6614#issuecomment-408745657

---

OK sorry again @livecodemichael . While I'm not sure I entirely agree with the LCS syntax because it seems more sensible to me to always request permission when starting to use something that needs it which will silently return granted if the user has already granted it and then just checking has permission to guard code later on I presume @runrevmark will want it the same for the same reasons.